tests/test_simulate.py

```diff
@@ -168,11 +168,13 @@ def test_simulate(model_setup):
 
     n_agents = 100_000
 
+    # breakpoint()
+
     # We need 64 because we do not alter the model array dtypes.
     initial_states = {
         "period": np.zeros(n_agents, dtype=np.int64),
         "lagged_choice": np.zeros(
-            n_agents, dtype=np.int64
+            n_agents, dtype=np.int32
```

Comment:
In introduced an error with this change. Our code needs to be robust against dtype specifications. Same in the other test. Here we probably need to fix the dtype of the decision draw to the dtype of the decisions